lua/telescope/config.lua

```diff
@@ -85,15 +89,58 @@ function config.set_defaults(defaults)
     - "cycle" (default)
     - "limit"]])
 
-  set("layout_strategy", "horizontal")
-  set("layout_defaults", {})
-
-  set("width", 0.75)
+  set("layout_strategy", "horizontal", [[
+    Determines the default layout of Telescope pickers.
+    See |telescope.layout| for details of the available strategies.
+
+    Default: 'horizontal']])
+  set("layout_config", {
+    horizontal = {
+      width = 0.8,
+      height = 0.9,
+    },
+    vertical = {
+      width = 0.8,
+      height = 0.9,
+    },
+    center = {
+      width = 0.5
+    },
+    bottom_pane = {
+      height = 0.5
+    },
+  },[[
+    Determines the default configuration values for layout strategies.
+    See |telescope.layout| for details of the configurations options for
+    each strategy.
+
+    Allows setting defaults for all strategies as top level options and
+    for overriding for specific options.
+    For example, the default values below set the default width to 80% of
+    the screen width for all strategies except 'center', which has width
+    of 50% of the screen width.
+
+    Default:
+    {
```

Comment:
we can we get this generated from the value above?

like `layou_config_default = { ... } `, and then use `vim.inspect` so that it doesn't get  out of date?

Reply:
I'm not sure what the best way to do this is.
Also, it _might_ make it easier for the explanation that goes along with it to get out of date.